google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineStandardDeployTask.java

```diff
@@ -37,16 +39,27 @@
 
   private AppEngineDeploy deploy;
   private AppEngineStandardStage stageStandard;
+  private boolean isFlexCompat;
 
+  /**
+   * @param isFlexCompat does not change any behavior of actual deployment. Provided solely
+   *     for the purpose of Analytics usage reporting.
+   */
   public AppEngineStandardDeployTask(
       @NotNull AppEngineDeploy deploy,
-      @NotNull AppEngineStandardStage stageStandard) {
+      @NotNull AppEngineStandardStage stageStandard,
+      boolean isFlexCompat) {
     this.deploy = deploy;
     this.stageStandard = stageStandard;
+    this.isFlexCompat = isFlexCompat;
   }
 
   @Override
   public void execute(ProcessStartListener startListener) {
+    UsageTrackerProvider.getInstance()
+        .trackEvent(GctTracking.CATEGORY, GctTracking.APP_ENGINE_DEPLOY,
+            isFlexCompat ? "flex.compat" : "standard", null);
```

Comment:
Hmm "flex-compat" isn't really an environment (we had an extended discussion about this in another PR). I almost prefer it the way you had it despite the overloading. But maybe @patflynn can give his opinion on this.